src/Command/UpdateRecipesCommand.php

```diff
@@ -0,0 +1,407 @@
+                sprintf('  <info>composer recipes:install %s --force -v</info>', $packageName),
+            ]);
+
+            return 1;
+        }
+
+        $originalRecipe = $this->getRecipe($packageName, $recipeRef, $recipeVersion);
+
+        if (null === $originalRecipe) {
+            $io->writeError([
+                'The original recipe version you have installed could not be found, it may be too old.',
+                'Update the recipe by re-installing the latest version with:',
+                sprintf('  <info>composer recipes:install %s --force -v</info>', $packageName),
+            ]);
+
+            return 1;
+        }
+
+        $newRecipe = $this->getRecipe($packageName);
+
+        if ($newRecipe->getRef() === $originalRecipe->getRef()) {
+            $io->write(sprintf('This recipe for <info>%s</info> is already at the latest version.', $packageName));
+
+            return 0;
+        }
+
+        $recipeUpdate = new RecipeUpdate($originalRecipe, $newRecipe, $symfonyLock, $this->rootDir);
+        $this->configurator->populateUpdate($recipeUpdate);
+        $originalComposerJsonHash = $this->flex->getComposerJsonHash();
+        $patcher = new RecipePatcher($this->rootDir, $io);
+
+        try {
+            $patch = $patcher->generatePatch($recipeUpdate->getOriginalFiles(), $recipeUpdate->getNewFiles());
+            $hasConflicts = !$patcher->applyPatch($patch);
+        } catch (\Throwable $throwable) {
+            $io->writeError([
+                '<bg=red;fg=white>There was an error applying the recipe update patch</>',
+                $throwable->getMessage(),
+                '',
+                'Update the recipe by re-installing the latest version with:',
+                sprintf('  <info>composer recipes:install %s --force -v</info>', $packageName),
+            ]);
+
+            return 1;
+        }
+
+        $symfonyLock->add($packageName, $newRecipe->getLock());
+        $this->flex->finish($this->rootDir, $originalComposerJsonHash);
+
+        $io->write([
+            '  <bg=blue;fg=white>                      </>',
+            '  <bg=blue;fg=white> Yes! Recipe updated! </>',
+            '  <bg=blue;fg=white>                      </>',
+            '',
+        ]);
+
+        if ($hasConflicts) {
+            $io->write([
+                '  The recipe was updated but with <bg=red;fg=white>one or more conflicts</>.',
+                '  Run <comment>git status</comment> to see them.',
+                '  After resolving, commit your changes like normal.',
+            ]);
+        } else {
+            if (!$patch->getPatch()) {
+                // no changes were required
+                $io->write([
+                    '  No files were changed as a result of the update.',
+                ]);
+            } else {
+                $io->write([
+                    '  Run <comment>git status</comment> the changes then commit like normal.',
```

Comment:
```suggestion
                    '  Run <comment>git status</comment> to see the changes then commit like normal.',
```

Reply:
Maybe also tell about `git diff --cached`. It was a bit unexpected to me when I ran the first time that this command added all changes to the index already.